File: game-5-sim/src/physics.rs
use crate::geometry::{Point, Radians};

pub type Velocity = f32;

pub fn move_point (left_top: &Point, velocity: &Velocity, direction: &Radians) -> Point {
  return Point {
    x: left_top.x + velocity * direction.cos(),
    y: left_top.y + velocity * direction.sin()
  }
}

File: packages/page-server/src/sim/physics.ts
import { Point, Radians } from "./geometry";

export type Velocity = number;

export function move_point(location: Point, velocity: Velocity, rotation: Radians): Point {
  return {
    x: location.x + velocity * Math.cos(rotation),
    y: location.y + velocity * Math.sin(rotation)
  }
}